### DISCUSSION TITLE
Release of ezdxf v1.3.5

### DISCUSSION THREAD

**Discussion by mozman** [mozman/ezdxf#1218] [Announcements]
## Version 1.3.5 - 2024-12-15

- [Release Notes v1.3.x](https://ezdxf.mozman.at/notes/#/page/release%20notes%20v1.3.x)
- NEW: `ezdxf.blkrefs.find_unreferenced_blocks()` function returns the names of unused block definitions
- NEW: `Auditor` checks if modelspace and active paperspace exists
- NEW: support for multi-line attributes for `Insert.add_auto_attribs()` method
    - contributed by @gswifort
- CHANGE: load `ODAFC` path dynamically from `ezdxf.options`
- CHANGE: load `OpenSCAD` path dynamically from `ezdxf.options`
- REMOVE: `_AbstractLayout.rename()`, unused method without implementation and documentation
- BUGFIX: `BlocksSection.delete_block()` will not delete layout blocks in safe mode.
- BUGFIX: #1203
    - copying dynamic blocks via clipboard removes dynamic features
- BUGFIX: #1215
    - measurement prefix was prepended to overridden text for radius and diameter dimensions

**Comment by Quasarrs** [comment]
I don’t quite understand how construction works using multi-line text in an attribute?

``` python
import ezdxf

doc = ezdxf.new(dxfversion="R2018")


block = doc.blocks.new(name="MY_BLOCK")
block.add_attdef("TAG1", (0, 0), "Default Value", height=0.25)
block.add_attdef("TAG2", (0, -1), "Another Default", height=0.25)


msp = doc.modelspace()
insert = msp.add_blockref("MY_BLOCK", insert=(5, 5))
attribs = {
    "TAG1": "Value\n 1",
    "TAG2": "Value\n 2",
}
insert.add_auto_attribs(attribs)
doc.saveas("example.dxf")
```
![image](https://github.com/user-attachments/assets/3421f810-eee0-4456-afc3-62cf3a96bb66)

**Comment by mozman** [comment]
Here is an example how to create a multiline ATTDEF:

```Python
import ezdxf

doc = ezdxf.new(dxfversion="R2018")
block = doc.blocks.new(name="MY_BLOCK")

# Text, location and height not needed, because copied from MTEXT entity
attdef = block.add_attdef("TAG1")
# Define the multiline attribute as MTEXT entity
mtext = block.add_mtext("Default Value", dxfattribs={"char_height": 0.25})
mtext.set_location((0, 0))
# Set ATTDEF content from MTEXT entity
attdef.set_mtext(mtext)

attdef = block.add_attdef("TAG2", (0, -1))
# reuse existing MTEXT entity
mtext.text = "Another Default"
mtext.set_location((0, -1))
# Set ATTDEF content from MTEXT entity and destroy the MTEXT entity
attdef.embed_mtext(mtext)

# Usage of add_auto_attribs() with multiline ATTDEFs:
msp = doc.modelspace()
insert = msp.add_blockref("MY_BLOCK", insert=(5, 5))
attribs = {
	"TAG1": "TAG1-Line1\nTAG1-Line2",
	"TAG2": "TAG2-Line3\nTAG2-Line4",
}
insert.add_auto_attribs(attribs)
doc.saveas("block_with_multiline_attdef.dxf")
```

![image](https://github.com/user-attachments/assets/dc58274a-63da-47ab-906a-e1d558b93049)

**Comment by Quasarrs** [comment]
Thank you very much for your efforts. Everything works, I will use it

**Comment by Quasarrs** [comment]
@mozman 
 ```python       
odafc.win_exec_path = (
  
          "C:\\Program Files\\ODA\\ODAFileConverter 24.10.0\\ODAFileConverter.exe"
        )
        if not odafc.is_installed():
```
```
  File "C:\Python_v\interpret\Tools_312\.venv\Lib\site-packages\ezdxf\addons\odafc.py", line 398, in _get_odafc_path
    raise ODAFCNotInstalledError(
ezdxf.addons.odafc.ODAFCNotInstalledError: Could not find ODAFileConverter in the path. Install application from https://www.opendesign.com/guestfiles/oda_file_converter
```
Check for installation of ODA File Converter stopped working in 1.3.5
I rolled back to 1.3.4 and everything works.

**Reply by mozman** [reply]
Use the global `options` object to set the path or use a config file. The global variable `win_exec_path` was removed.

**Reply by mozman** [reply]
```Python
ezdxf.options.set("odafc-addon", "win_exec_path", "your-path")
```
